### ISSUE TITLE
Opacity mask in CanvasPage broken using DeferredRenderer

### ISSUE THREAD

**Issue by nc4rrillo** [AvaloniaUI/Avalonia#1693]
The blue rect in the Canvas Page is supposed to render using an opacity mask for a fade effect. This is broken using deferred and works using the immediate renderer.

Expected:
![image](https://user-images.githubusercontent.com/17953668/41825451-47917034-77ee-11e8-8c32-cb4dd2018f64.png)

Actual:
![image](https://user-images.githubusercontent.com/17953668/41825462-7c301cf0-77ee-11e8-91cb-ca8c98445d15.png)

Curiously, setting the Start/Stop endpoints to be absolute units fixes the problem.

**Comment by nc4rrillo** [comment]
cc: @grokys

**Comment by nc4rrillo** [comment]
This is due to `PushOpacityMask` in `DeferredRenderer` using `ClipBounds` vs `Bounds` in `ImmediateRenderer`. 

If I set `ClipToBounds` on the `Rectangle` visual it works as expected. If I change this to `Bounds` it also works but it not until I resize the window. Seems like some invalidation issue.

**Comment by danwalmsley** [comment]
@nc4rrillo hey dude how to I contact you?
